akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -780,12 +780,27 @@ abstract class GraphStageLogic private[stream] (val inCount: Int, val outCount:
       setHandler(out, previous)
       andThen()
       if (followUps != null) {
-        getHandler(out) match {
-          case e: Emitting[_] ⇒ e.as[T].addFollowUps(this)
```

Comment:
This line caused corruption, because it prepended queue of elements to be emitted with new element and it was pushed before the elements that had come before